File: python/meteroid/models/applied_coupon.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime
from decimal import Decimal

from ..serialization import BaseModel
from .applied_coupon_id import AppliedCouponId
from .coupon_id import CouponId


@dataclasses.dataclass
class AppliedCoupon(BaseModel):
    coupon_id: CouponId

    created_at: datetime

    id: AppliedCouponId

    is_active: bool

    applied_amount: t.Optional[Decimal] = None

    applied_count: t.Optional[int] = None

    last_applied_at: t.Optional[datetime] = None

File: go/applied_coupon.go
// this file is @generated
package meteroid

import "time"

type AppliedCoupon struct {
	AppliedAmount *string `json:"applied_amount,omitempty"`

	AppliedCount *int32 `json:"applied_count,omitempty"`

	CouponId CouponId `json:"coupon_id"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Id AppliedCouponId `json:"id"`

	IsActive bool `json:"is_active"`

	// RFC 3339 timestamp.
	LastAppliedAt *time.Time `json:"last_applied_at,omitempty"`
}
